packages/flutter_tools/lib/src/commands/format.dart

```diff
@@ -40,26 +41,36 @@ class FormatCommand extends FlutterCommand {
       'format',
     ];
     final List<String> rest = argResults?.rest ?? <String>[];
-    if (rest.isEmpty) {
-      globals.printError(
-        'No files specified to be formatted.'
-      );
+    if (rest.isEmpty || onlyHasOptions(rest)) {
+      globals.printError('No files specified to be formatted.');
       command.add('-h');
     } else {
       command.addAll(<String>[
         for (String arg in rest)
-          if (arg == '--dry-run' || arg == '-n')
-            '--output=none'
-          else
-            arg,
+          if (arg == '--dry-run' || arg == '-n') '--output=none' else arg,
       ]);
     }
 
-    final int result = await globals.processUtils.stream(command);
-    if (result != 0) {
-      throwToolExit('Formatting failed: $result', exitCode: result);
+    final RunResult result = globals.processUtils.runSync(command);
+    if (result.exitCode != 0) {
+      throwToolExit('Formatting failed: ${result.stderr}', exitCode: result.exitCode);
+    } else {
+      globals.logger.printStatus(result.stdout);
     }
 
     return FlutterCommandResult.success();
   }
+
+  bool onlyHasOptions(List<String> allOptions) {
```

Comment:
Shouldn't this instead check all arguments and return true if any do NOT start with both `--` and `-`? Consider the following arguments: `['--fix', './lib/main.dart']`

Also, I think the parameter should be named `allArguments`, since it may not be an option.

Reply:
Also can you document why we want to check if the argument only has options (and why `-h` doesn't count as an option)

Reply:
The way I choose to do it is to only return `True` if all the arguments are options therefore the `onlyHasOptions` name; in your example one argument is not an option so this will return `False` and we will not hit the error message `globals.printError('No files specified to be formatted.');`.
About the -h sure I can add some comments (and rename `allOptions` to `arguments` I think that way it will be more clear what this is doing).

Reply:
ahh I see, yeah, you're right.

Reply:
actually, can you specifically check for `-h` or `--help` before calling `onlyHasOptions`, because if they enter `flutter format -h` we don't want to print `'No files specified to be formatted.'`?

Reply:
Something like:

```dart
    if (rest.length == 1 && rest.first == '-h' || rest.first == '--help') {
      command.add('-h');
    } else if (rest.isEmpty || onlyHasOptions(rest)) {
      globals.printError('No files specified to be formatted.');
      command.add('-h');
    } else {
      command.addAll(<String>[
        for (String arg in rest)
          if (arg == '--dry-run' || arg == '-n') '--output=none' else arg,
      ]);
    }
```

Reply:
Yeah we don't want that. That is why it returns False (and don't go to the error message) when -h is present but I agree on moving it out of the function and do the check before

Reply:
Ahh yeah, you're right again. Time for a second cup of coffee?